src/mmg3d/hash_3d.c

```diff
@@ -1837,10 +1825,71 @@ int MMG5_chkBdryTria(MMG5_pMesh mesh) {
     fprintf(stderr,"\n  ## Warning: %s: %" MMG5_PRId " extra boundaries founded\n",
             __func__,nbl);
 
-  /* Fill missing bdy triangles */
-  return MMG5_bdryTria(mesh,ntmesh);
 }
 
+int MMG5_chkbdryTria_deleteExtraBoundaries() {
+  nt=0; nbl=1;
+
+  if ( ! MMG5_hashNew(mesh,&hashTri,0.51*mesh->nt,1.51*mesh->nt) ) return 0;
+
+  for (k=1; k<=mesh->nt; k++) {
+    ptt = &mesh->tria[k];
+
+    ia = ptt->v[0];
+    ib = ptt->v[1];
+    ic = ptt->v[2];
+
+    i = MMG5_hashGetFace(&hashElt,ia,ib,ic);
+    j = MMG5_hashFace(mesh,&hashTri,ia,ib,ic,k);
+
+    ptt->cc = i;
+
+    if ( !j ) {
+      MMG5_DEL_MEM(mesh,hashElt.item);
+      MMG5_DEL_MEM(mesh,hashTri.item);
+      return 0;
+    }
+    else if ( j > 0 ) {
+      /* the face already exists in the tria table */
+      continue;
```

Comment:
I think that here we can insert the tria deletion in order to perform the compression of triangle array outside the loop (see comment where the array is currently packed).

With something like : ptt->v[0] = 0; MG_EOK(ptt) should be false.